src/smc-webapp/project_actions.ts

```diff
@@ -2803,6 +2803,8 @@ export class ProjectActions extends Actions<ProjectStoreState> {
       cb: err => {
         this.fetch_directory_listing();
         this.set_activity({ id, stop: "" });
+        this.setState({ downloading_file: false });
+        this.set_active_tab("files", { update_file_listing: false });
```

Comment:
Change downloading a file so that it renders a spinner until it's done downloading.